### PR TITLE
Hydro 2556 - /oauth/authorize

### PR DESCRIPTION
**Pull request by manioc2015**

### CODE REVIEW

File: components/schemas.yaml [hunk 1]
```diff
@@ -830,6 +830,30 @@ OAuthRegisterRequest:
...
+          client_id:
+            type: string
+          callback_url:
+            type: string
+          response_type:
+            type: string
+          scope:
+            type: string
+          state:
+            type: string
```
Comment (kulyim) [inline components/schemas.yaml]: I don't think we need this. This is defined in the oauth spec and conformant clients will already be using the correct parameter names.

File: api.yaml [hunk 2]
```diff
@@ -14,6 +14,18 @@ info:
 servers:
   - url: https://virtserver.swaggerhub.com/PhotoShelter/PhotoShelter/4
     description: PhotoShelter Auto Mocking API
+components:
+  securitySchemes:
+    OAuthSecurity:
```
Comment (kulyim) [inline api.yaml]: This is good. This defines a security scheme named "OAuthSecurity" that can be added to all or some routes to signal the route requires an oauth token and some (possibly optional) scope.

File: components/responses.yaml [hunk 3]
```diff
@@ -236,6 +236,14 @@ OAuthRegisterResponse:
...
             $ref: ./schemas.yaml#/OAuthRegister
 
+OAuthAuthorizeResponse:
+  description: 'OAuth Authorize Response'
+  headers:
+    Location:
+      schema:
+        type: string
+      description: OAuth Authorization Callback URL
+
```
Comment (kulyim) [inline components/responses.yaml]: I don't think we need this

File: paths/oAuthAuthorize.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  operationId: oAuthAuthorize
+  requestBody:
+    $ref: ../components/schemas.yaml#/OAuthAuthorizeRequest
+  responses:
+    302:
+      $ref: ../components/responses.yaml#/OAuthAuthorizeResponse
+    400:
+      $ref: ../components/errors.yaml#/BadRequestErrorResponse
+    401:
+      $ref: ../components/errors.yaml#/UnauthorizedErrorResponse
```
Comment (kulyim) [inline paths/oAuthAuthorize.yaml]: We don't need this.

**Comment by kulyim** [comment]
We should define at least two security schemes
 1 ApiAuthKey
 2 OAuthSecurity
The first will cover /login /register and other endpoints that require membership but not authorization. The second security scheme will cover all the endpoints that need authorization.
The toplevel "security" keyword configures the security specification for the entire api. Each operation/route can override the global security specification (see example below) to configure that particular operation/route. In our case, OAuthSecurity with whatever scope applies to the operation/route.


```
security:
    - ApiAuthKey
paths:
   /get-the-good-resource
      get:
      summary: blah blah
      security:
         - OAuthSecurity: [write:all]
         responses:
            '200': ...
```

Thoughts?

File: components/responses.yaml [hunk 5]
```diff
@@ -235,7 +235,7 @@ OAuthRegisterResponse:
         properties:
           data:
             $ref: ./schemas.yaml#/OAuthRegister
-
+
```
Comment (yourshainess) [inline components/responses.yaml]: Remove these whitespace changes that is unnecessary

File: components/schemas.yaml [hunk 6]
```diff
@@ -830,7 +830,7 @@ OAuthRegisterRequest:
             type: string
           logo_uri:
             type: string
-
+
```
Comment (yourshainess) [inline components/schemas.yaml]: Remove these whitespace changes that is unnecessary

File: api.yaml [hunk 7]
```diff
@@ -12,8 +12,28 @@ info:
     url: https://www.photoshelter.com/support/license
   version: '1.0'
 servers:
-  - url: https://virtserver.swaggerhub.com/PhotoShelter/PhotoShelter/4
+  - url: /psapi/v4.0
```
Comment (yourshainess) [inline api.yaml]: What's the reason behind this change?

Comment (manioc2015) [reply api.yaml]: we're not using swagger anymore so i thought we should update the server url
